src/main/java/com/labs/introtoprogramming/lab4/image/io/bmp/BMPParser.java

```diff
@@ -0,0 +1,66 @@
+package com.labs.introtoprogramming.lab4.image.io.bmp;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.Arrays;
+
+class BMPParser {
+
+  int parseHeader(byte[] headerInfo) throws UnsupportedFileFormat {
+    if (headerInfo.length != 14) {
+      throw new UnsupportedFileFormat("File format is not a BMP");
+    }
+    if (headerInfo[0] != 0x42 || headerInfo[1] != 0x4D) {
+      throw new UnsupportedFileFormat("File format is not a BMP");
+    }
+
+    int size = sumUpBytes(headerInfo, 2, 6);
+    int offset = sumUpBytes(headerInfo, 10, 14);
+    if (offset < 54 || offset > size) {
+      throw new UnsupportedFileFormat("Invalid value for offset");
+    }
+    return offset;
+  }
+
+  BMPImageHeader parseImageHeader(byte[] headerInfo) throws UnsupportedFileFormat {
+    int headerSize = sumUpBytes(headerInfo, 0, 4);
+    if (headerSize != headerInfo.length) {
+      throw new UnsupportedFileFormat("Incorrect image header");
```

Comment:
Again, I am not sure if `ImageLoadException` is a right choice here. Maybe `ImageLoadException` or something else?